_source/_posts/2022-03-15-react-vite-number-converter.md

```diff
@@ -0,0 +1,350 @@
+
+```bash
+npm install -E @okta/okta-react@6.4.2 @okta/okta-auth-js@6.1.0
+```
+
+Vite needs some help resolving import aliases used by the Okta libraries. To make things work, open `vite.config.ts` and add the following entry to the configuration object.
+
+```ts
+resolve: {
+  alias: [
+    {
+      find: "@okta/okta-auth-js",
+      replacement: require.resolve("@okta/okta-auth-js/dist/okta-auth-js.umd.js"),
+    },
+  ],
+}
+```
+
+If your IDE or build can't resolve `require`, you may need to add the `@types/node` library for the `require` method. In the terminal, add the library by running the following command:
+
+```shell
+npm i –save-dev @types/node
+```
+
+Now, open `src/main.tsx` and add the following code immediately after the `import` statements.
+
+```ts
+import { Security } from "@okta/okta-react";
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
+
+const oktaAuth = new OktaAuth({
+  issuer: `https://{yourOktaDomain}/oauth2/default`,
+  clientId: '{yourClientID}',
+  redirectUri: `${window.location.origin}/callback`,
+});
+
+function restoreOriginalUri(oktaAuth: OktaAuth, originalUri: string) {
+  window.location.replace(
+    toRelativeUrl(originalUri || "/", window.location.origin)
+  );
+}
+```
+
+Here `{yourClientID}` is the client ID that you obtained earlier and `{yourOktaDomain}` is your Okta domain. Next, surround the `BrowserRouter` component with the `Security` component, passing in `oktaAuth` and `restoreOriginalUri` as parameters. The call to the render function should look something like this.
+
+```ts
+ReactDOM.render(
+  <React.StrictMode>
+    <Security oktaAuth={oktaAuth} restoreOriginalUri={restoreOriginalUri}>
+      <BrowserRouter>
+        <App />
+      </BrowserRouter>
+    </Security>
+  </React.StrictMode>,
+  document.getElementById('root')
+)
+```
+
+Now that you have made the `OktaAuth` object available to your application, you are ready to create a secured route and configure the login process. Open `src/App.tsx` and add the following import.
+
+```ts
+import { LoginCallback, SecureRoute } from '@okta/okta-react';
+```
+
+Finally, convert the route that serves the `Converter` component to a secured route and add another route that handles the login callback from Okta. The code inside the `Switch` component should resemble the code below.
+
+```ts
+<Switch>
+  <Route exact path="/" component={Home} />
+  <SecureRoute path="/converter" component={Converter} />
+  <Route path="/callback" component={LoginCallback} />
+</Switch>
+```
+
+Congratulations, you are ready to power up the application again. If it is still not running, run the command below in the terminal.
+
+```bash
+npm run dev
+```
+
+Now, whenever you navigate to the **Converter** route, the application checks if you are authenticated. If not, it will redirect you to the Okta sign-in page, where you can enter your user credentials. After successfully logging in, you are to the application page you requested.
```

Comment:
```suggestion
Now, whenever you navigate to the **Converter** route, the application checks if you are authenticated. If not, it will redirect you to the Okta sign-in page, where you can enter your user credentials. After successfully logging in, you are able to access the application page you requested.
```
something missing